.github/workflows/test.yml

```diff
@@ -31,22 +27,20 @@ jobs:
         uses: golangci/golangci-lint-action@v3
         with:
           version: latest
+
   test:
     runs-on: ubuntu-latest
-    container:
-      image: golang:1.19-alpine
-
     steps:
-      - name: Install Dependencies
-        run: apk add --no-cache git gcc g++ curl bash tar
-
-      - name: GIT permission workaround https://github.blog/2022-04-12-git-security-vulnerability-announced/
-        run: git config --global --add safe.directory /__w/saucectl/saucectl
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19
```

Comment:
I'd suggest to use `actions/setup-go` with caching. Then you can skip the use of `actions/cache` below.

```suggestion
          go-version: 1.19
          cache: true
```